runtime/compiler/p/codegen/PPCPrivateLinkage.cpp

```diff
@@ -1913,6 +1913,42 @@ int32_t TR::PPCPrivateLinkage::buildPrivateLinkageArgs(TR::Node
 	    addDependency(dependencies, NULL, (TR::RealRegister::RegNum)i, TR_VSX_SCALAR, cg());
       }
 
+
+
+   /*
+   
+   PPC ABI --
+   
+   AIX 32-bit BE:
+   AIX 64-bit BE:
+   Linux 64-bit BE: 
+   These are aix-style linkage. 
+   There is a TOC-base register (gr2) from function descriptor if that function is called. 
+   If the callee function is in libj9jitxx.so module, its TOC-base is recorded in J9VMThread jitTOC field already. 
+   All helper functions are in libj9jitxx.so module. So, you can load it from jitTOC field of J9VMThread (pointed to by metaDataRegister)
+  
+   Linux 32bit BE: Doesn't contain gr2. The register is reserved.
+
+   Linux 64-bit LE: there is TOC (gr2), but there is no function descriptor. 
+   TOC-base set-up is defined by ABI to go through a function's global-entry (relocation). 
+   The global-entry address should be set up in gr12. (although TOC-base is recorded in jitTOC field of J9VMThread as well ... none-C helpers are using it).
+
+   */
+   
+   if(linkage == TR_CHelper)
+   {
+      if(TR::Compiler->target.isLinux() && TR::Compiler->target.is64Bit() && TR::Compiler->target.cpu.isLittleEndian())
+      {
+         uintptrj_t targetAddress = (uintptrj_t)callSymbol->getMethodAddress();
+         loadConstant(cg(), callNode, (int64_t)targetAddress,dependencies->searchPreConditionRegister(TR::RealRegister::gr12));
+      }
+      else if (TR::Compiler->target.isAIX() || (TR::Compiler->target.isLinux() && TR::Compiler->target.is64Bit()))
+      {
+         generateTrg1MemInstruction(cg(),TR::InstOpCode::Op_load, callNode, dependencies->searchPreConditionRegister(TR::RealRegister::gr2), 
```

Comment:
Why are we sure that searchPreConditionRegister(TR::RealRegister::gr2) will not return NULL? Apparently it was tested and works somehow, but I am not sure how.

Reply:
The method searchPreConditionRegister essentially just checks if the register was added as a dependency. It returns the register if it was added, and null if it was never added as a dependency. 

We can see in buildPrivateLinkageArgs that in the case of CHelpers it always is added to the dependencies. In the case of Linux x32, it isn't since getPreserved will return True. 

```
   if (!properties.getPreserved(TR::RealRegister::gr2))
      {
      // Helper linkage preserves all registers that are not argument registers, so we don't need to spill them.
      if (linkage != TR_Helper)
         addDependency(dependencies, NULL, TR::RealRegister::gr2, TR_GPR, cg());
      }
```

Reply:
Thanks. Adding a brief comment or even an ASSERT could be helpful.